pandas/io/excel/_openpyxl.py

```diff
@@ -506,10 +512,23 @@ def _write_cells(
                 for k, v in style_kwargs.items():
                     setattr(xcell, k, v)
 
+            abs_row = startrow + cell.row + 1
+            abs_col = startcol + cell.col + 1
+
+            # track bounds (1-based for openpyxl)
+            if min_r is None or abs_row < min_r:
```

Comment:
Are we able to determine these by `startrow` / `startcol`  along with the DataFrame shape?